src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -3006,120 +2431,34 @@ ss::future<checked<tx_metadata, tx::errc>> tx_gateway_frontend::get_latest_tx(
       tx_id,
       pid,
       term);
-    auto r0 = co_await get_tx(term, stm, tx_id, timeout);
-    if (!r0.has_value()) {
-        co_return r0.error();
+    auto tx_result = co_await find_and_try_progressing_transaction(
```

Comment:
`find_and_try_progressing_transaction` - the method name is 🔥 the code is so easy to follow now.